### PR TITLE
fix: install packages with incompatible libc

### PR DESCRIPTION
**Pull request by KSXGitHub**
Fixes https://github.com/pnpm/pnpm/issues/7362

### CODE REVIEW

File: pkg-manager/plugin-commands-installation/src/install.ts [hunk 1]
```diff
@@ -325,6 +325,9 @@ export async function handler (opts: InstallCommandOptions): Promise<void> {
     devDependencies: opts.dev !== false,
     optionalDependencies: opts.optional !== false,
   }
+  // npm registry's abbreviated metadata currently does not contain libc
+  // see <https://github.com/pnpm/pnpm/issues/7362#issuecomment-1971964689>
+  const fetchFullMetadata = opts.rootProjectManifest?.pnpm?.supportedArchitectures?.libc && true
```
Comment (KSXGitHub) [inline pkg-manager/plugin-commands-installation/src/install.ts]: Is this better?

```suggestion
  const fetchFullMetadata = opts.rootProjectManifest?.pnpm?.supportedArchitectures?.libc ? true : null
```

File: store/store-connection-manager/src/createNewStoreController.ts [hunk 2]
```diff
@@ -60,7 +62,7 @@ export async function createNewStoreController (
     cacheDir: opts.cacheDir,
     cert: opts.cert,
     fullMetadata,
-    filterMetadata: fullMetadata,
+    filterMetadata,
```
Comment (zkochan) [inline store/store-connection-manager/src/createNewStoreController.ts]: you should filter metadata always. Just update the code that filters to keep the libc field.

Comment (KSXGitHub) [reply store/store-connection-manager/src/createNewStoreController.ts]: It does mean that the bug won't immediately be fixed unless the user clears the old metadata cache.

Comment (zkochan) [reply store/store-connection-manager/src/createNewStoreController.ts]: you can bump the version in FULL_FILTERED_META_DIR

File: pkg-manager/plugin-commands-installation/src/installDeps.ts [hunk 3]
```diff
@@ -116,6 +116,7 @@ export type InstallDepsOptions = Pick<Config,
   workspace?: boolean
   includeOnlyPackageFiles?: boolean
   prepareExecutionEnv: PrepareExecutionEnv
+  fetchFullMetadata?: boolean | null
```
Comment (zkochan) [inline pkg-manager/plugin-commands-installation/src/installDeps.ts]: why is null needed? This already support `undefined`.
```suggestion
  fetchFullMetadata?: boolean
```

Comment (KSXGitHub) [reply pkg-manager/plugin-commands-installation/src/installDeps.ts]: I think [`? true : null`](https://github.com/pnpm/pnpm/pull/8569/commits/268a242863c4ffb0e13798b0c3b695ff78424f89) would be clearer than `?? true`.

Comment (zkochan) [reply pkg-manager/plugin-commands-installation/src/installDeps.ts]: why would you use `?? true`? That wouldn't work correctly. You'd use `? true : undefined`

Comment (KSXGitHub) [reply pkg-manager/plugin-commands-installation/src/installDeps.ts]: I'm sorry, I meant `&& true`. (I got the symbol mixed up)

Comment (zkochan) [reply pkg-manager/plugin-commands-installation/src/installDeps.ts]: ok, well use `? true : undefined`. That is less confusing.

Comment (KSXGitHub) [reply pkg-manager/plugin-commands-installation/src/installDeps.ts]: I chose instead to explicitly specify the type as `true | undefined` (https://github.com/pnpm/pnpm/pull/8569/commits/7cd1b894c71b328dff9977f7ef3b388ad0407e61).

I think `? true : undefined` would risk future me going back and do refactor or even somebody even change it to actual boolean due to misunderstanding `underfined` as `false`.

**Review by zkochan — APPROVED** [review]

**Pull request merged**